components/brave_ads/core/internal/user_engagement/site_visit/site_visit_util.cc

```diff
@@ -5,12 +5,55 @@
 
 #include "brave/components/brave_ads/core/internal/user_engagement/site_visit/site_visit_util.h"
 
+#include "base/notreached.h"
+#include "base/types/cxx23_to_underlying.h"
 #include "brave/components/brave_ads/core/internal/common/url/url_util.h"
+#include "brave/components/brave_ads/core/internal/settings/settings.h"
 #include "brave/components/brave_ads/core/internal/tabs/tab_manager.h"
+#include "brave/components/brave_ads/core/mojom/brave_ads.mojom.h"
+#include "brave/components/brave_ads/core/public/ads_feature.h"
 #include "url/gurl.h"
 
 namespace brave_ads {
 
+bool IsAllowedToLandOnPage(const mojom::AdType mojom_ad_type) {
+  switch (mojom_ad_type) {
+    case mojom::AdType::kInlineContentAd:
+    case mojom::AdType::kPromotedContentAd: {
+      return UserHasOptedInToBraveNewsAds();
+    }
+
+    case mojom::AdType::kNewTabPageAd: {
+      // Only if:
+      // - The user has opted into new tab page ads and has either joined Brave
+      //   Rewards or new tab page ad events should always be triggered.
+      return UserHasOptedInToNewTabPageAds() &&
+             (UserHasJoinedBraveRewards() ||
+              ShouldAlwaysTriggerNewTabPageAdEvents());
+    }
+
+    case mojom::AdType::kNotificationAd: {
+      return UserHasOptedInToNotificationAds();
+    }
+
+    case mojom::AdType::kSearchResultAd: {
+      // Only if:
+      // - The user has opted into search result ads and has either joined Brave
+      //   Rewards or search result ad events should always be triggered.
+      return UserHasOptedInToSearchResultAds() &&
+             (UserHasJoinedBraveRewards() ||
+              ShouldAlwaysTriggerSearchResultAdEvents());
```

Comment:
will this result in sending of search result ad landed confirmations for non-reward users?